tests/e2e/test_rosacli_cluster.go

```diff
@@ -2248,3 +2253,159 @@ var _ = Describe("Reusing opeartor prefix and oidc config to create clsuter", la
+			By("Delete cluster")
+			rosaClient.Runner.UnsetArgs()
+			_, err := clusterService.DeleteCluster(clusterID, "-y")
+			Expect(err).To(BeNil())
+
+			rosaClient.Runner.UnsetArgs()
+			err = clusterService.WaitClusterDeleted(clusterID, 3, 30)
+			Expect(err).To(BeNil())
+
+			By("Delete operator-roles")
+			_, err = ocmResourceService.DeleteOperatorRoles(
+				"-c", clusterID,
+				"--mode", "auto",
+				"-y",
+			)
+			Expect(err).To(BeNil())
+
+			By("Clean resource")
+			errs := profilehandler.DestroyResourceByProfile(customProfile, rosaClient)
+			Expect(len(errs)).To(Equal(0))
+		})
+
+		It("Creating cluster with sts external id should succeed - [id:75603]",
+			labels.Medium, labels.Runtime.Day1Supplemental,
+			func() {
+				By("Create classic cluster in auto mode")
+				testingClusterName = "rosa75603"
+				testOperatorRolePrefix := "rosa75603opp"
+				flags, err := profilehandler.GenerateClusterCreateFlags(customProfile, rosaClient)
+				Expect(err).ToNot(HaveOccurred())
+
+				command := "rosa create cluster --cluster-name " + testingClusterName + " " + strings.Join(flags, " ")
+				rosalCommand := config.GenerateCommand(command)
+				rosalCommand.ReplaceFlagValue(map[string]string{
+					"--operator-roles-prefix": testOperatorRolePrefix,
+				})
+
+				rosalCommand.AddFlags("--mode", "auto")
+
+				By("Update installer role")
+				ExternalId := "223B9588-36A5-ECA4-BE8D-7C673B77CEC1"
+				installRoleArn := rosalCommand.GetFlagValue("--role-arn", true)
+				_, roleName, err := common.ParseRoleARN(installRoleArn)
+				Expect(err).To(BeNil())
+
+				awsClient, err := aws_client.CreateAWSClient("", "")
+				Expect(err).To(BeNil())
+				opRole, err := awsClient.IamClient.GetRole(
+					context.TODO(),
+					&iam.GetRoleInput{
+						RoleName: &roleName,
+					})
+				Expect(err).To(BeNil())
+
+				decodedPolicyDocument, err := url.QueryUnescape(*opRole.Role.AssumeRolePolicyDocument)
+				Expect(err).To(BeNil())
+
+				By("update the trust relationship")
+				var policyDocument map[string]interface{}
+
+				err = json.Unmarshal([]byte(decodedPolicyDocument), &policyDocument)
+				Expect(err).To(BeNil())
+
+				newCondition := map[string]interface{}{
+					"StringEquals": map[string]interface{}{
+						"sts:ExternalId": ExternalId,
+					},
+				}
+
+				statements := policyDocument["Statement"].([]interface{})
+				for _, statement := range statements {
+					stmt := statement.(map[string]interface{})
+					stmt["Condition"] = newCondition
+				}
+				updatedPolicyDocument, err := json.Marshal(policyDocument)
+				Expect(err).To(BeNil())
+
+				_, err = awsClient.IamClient.UpdateAssumeRolePolicy(context.TODO(), &iam.UpdateAssumeRolePolicyInput{
+					RoleName:       aws.String(roleName),
+					PolicyDocument: aws.String(string(updatedPolicyDocument)),
+				})
+				Expect(err).To(BeNil())
+
+				By("Create cluster with external id not same with the role setting one")
+				notMatachExternalId := "333B9588-36A5-ECA4-BE8D-7C673B77CDCD"
```

Comment:
please fix this typo `notMatachExternalId` --> `notMatchExternalId` or `unMatchExternalId`